pkg/strategy/dca2/strategy.go

```diff
@@ -191,3 +229,19 @@ func (s *Strategy) updateTakeProfitPrice() {
 	s.takeProfitPrice = s.Market.TruncatePrice(s.Position.AverageCost.Mul(fixedpoint.One.Add(takeProfitRatio)))
 	s.logger.Infof("[DCA] cost: %s, ratio: %s, price: %s", s.Position.AverageCost, takeProfitRatio, s.takeProfitPrice)
 }
+
+func (s *Strategy) Close(ctx context.Context) error {
+	s.logger.Infof("[DCA] closing %s dca2", s.Symbol)
+
+	defer s.EmitClosed()
+
+	bbgo.Sync(ctx, s)
```

Comment:
you need to cancel the orders fast, and then run Sync

because the lag might cause partially fill